reference/ast-examples.md

```diff
@@ -0,0 +1,147 @@
+# AST Element Examples
+
+When building recipes, it's important to understand how certain elements in the OpenRewrite Abstract Syntax Tree (AST) correspond to code. You couldn't, for example, properly rename a variable with a recipe unless you knew that `J.Identifier` is how variable names are represented in the AST.
```

Comment:
I note the term "element" is used extensively throughout this document, but "element" isn't a term we define anywhere.

I wonder if using this term repeatedly may be confusing. It feels like it implies some formal distinction or concept that isn't there. "not complete/total" or "component of" is what I usually mean by "element" when I say "AST element", but readers might be wondering if it means something more formal. Is a `J.CompilationUnit` an "element" of an AST since it is a whole AST?

Reply:
Part of the language trickiness here is than an AST element is itself an AST. So in conversation we normally say "AST" to mean "the whole AST" and "AST element" to mean "any AST which is itself a component of another, larger AST"

Reply:
This is an interesting distinction. I think I have been guilty in the past of calling them "elements of an AST". Are you suggesting replacing `element` with `ast`? I think this makes sense the more I think about it...

"AST Element Examples" could just be "AST Examples"..."under how certain ASTs in the ....." etc.